src/Faithlife.Analyzers/OverloadWithStringComparisonAnalyzer.cs

```diff
@@ -57,90 +57,36 @@ public override void Initialize(AnalysisContext context)
-					methodSymbol.Parameters[1].Type.SpecialType == SpecialType.System_String &&
-					methodSymbol.Parameters[2].Type.Equals(stringComparisonType);
-
-			case var t when t == (false, "IndexOf", 1) || t == (false, "LastIndexOf", 1):
-				// string.IndexOf(char)
-				return methodSymbol.Parameters[0].Type.SpecialType == SpecialType.System_Char;
-
-			case var t when t == (false, "IndexOf", 2) || t == (false, "LastIndexOf", 2):
-				// string.IndexOf(char, int); string.IndexOf(string, StringComparison)
-				return (methodSymbol.Parameters[0].Type.SpecialType == SpecialType.System_Char && methodSymbol.Parameters[1].Type.SpecialType == SpecialType.System_Int32) ||
-						(methodSymbol.Parameters[0].Type.SpecialType == SpecialType.System_String && methodSymbol.Parameters[1].Type.Equals(stringComparisonType));
-
-			case var t when t == (false, "IndexOf", 3) || t == (false, "LastIndexOf", 3):
-				// string.IndexOf(char, int, int); string.IndexOf(string, int, StringComparison)
-				return (methodSymbol.Parameters[0].Type.SpecialType == SpecialType.System_Char &&
-					methodSymbol.Parameters[1].Type.SpecialType == SpecialType.System_Int32 &&
-					methodSymbol.Parameters[2].Type.SpecialType == SpecialType.System_Int32) ||
-					(methodSymbol.Parameters[0].Type.SpecialType == SpecialType.System_String &&
-					methodSymbol.Parameters[1].Type.SpecialType == SpecialType.System_Int32 &&
-					methodSymbol.Parameters[2].Type.Equals(stringComparisonType));
-
-			case var t when t == (false, "IndexOf", 4) || t == (false, "LastIndexOf", 4):
-				// string.IndexOf(string, int, int, StringComparison)
-				return methodSymbol.Parameters[0].Type.SpecialType == SpecialType.System_String &&
-					methodSymbol.Parameters[1].Type.SpecialType == SpecialType.System_Int32 &&
-					methodSymbol.Parameters[2].Type.SpecialType == SpecialType.System_Int32 &&
-					methodSymbol.Parameters[3].Type.Equals(stringComparisonType);
+				if (parameter.Type.SpecialType == SpecialType.System_Object)
+					return true;
+				if (parameter.Type.SpecialType == SpecialType.System_Char)
+					return true;
+				if (parameter.Type.Equals(stringComparisonType))
+					return true;
+				if (parameter.Type.Equals(cultureInfoType))
+					return true;
 			}
+			
 			return false;
 		}
 
-		static readonly DiagnosticDescriptor s_useStringComparisonRule = new DiagnosticDescriptor(
+		private static readonly ISet<string> s_affectedMethods = new HashSet<string>(new[] { "Equals", "Compare", "IndexOf", "LastIndexOf", "StartsWith", "EndsWith" });
+
+		private static readonly DiagnosticDescriptor s_useStringComparisonRule = new DiagnosticDescriptor(
 			id: UseStringComparisonDiagnosticId,
-			title: "Use StringComparison overload",
-			messageFormat: "Use an overload that takes a StringComparison.",
+			title: "Use StringComparison or CultureInfo overload",
+			messageFormat: "Use an overload that takes a StringComparison or CultureInfo.",
```

Comment:
Not sure I want to promote the `CultureInfo` overloads explicitly; they're fairly esoteric.

Reply:
Omitted.